zerver/views/messages.py

```diff
@@ -576,17 +580,23 @@ def ok_to_include_history(narrow: Optional[Iterable[Dict[str, Any]]], user_profi
 
     return include_history
 
-def get_stream_name_from_narrow(narrow: Optional[Iterable[Dict[str, Any]]]) -> Optional[str]:
+def get_stream_name_from_narrow(narrow: Optional[Iterable[Dict[str, Any]]], realm: Realm) -> Optional[str]:
+    def get_stream_name(operand: Union[str, int]) -> str:
+        if isinstance(operand, str):
+            return operand.lower()
+
+        return get_stream_by_id_in_realm(operand, realm).name.lower()
```

Comment:
If you read where this is used:
```
def exclude_muting_conditions(user_profile: UserProfile,                                  
                              narrow: Optional[Iterable[Dict[str, Any]]]) -> List[Selecta\
ble]:                                                                                     
    conditions = []                                                                       
    stream_name = get_stream_name_from_narrow(narrow, user_profile.realm)                 
                                                                                          
    stream_id = None                                                                      
    if stream_name is not None:                                                           
        try:                                                                              
            # Note that this code works around a lint rule that                           
            # says we should use access_stream_by_name to get the                         
            # stream.  It is okay here, because we are only using                         
            # the stream id to exclude data, not to include results.                      
            stream_id = get_stream(stream_name, user_profile.realm).id                    
        except Stream.DoesNotExist:                                                       
            pass                                                                          
```

I think we want this to be `get_stream_from_narrow` returning a `Stream` object.

Also, maybe we should have a prep commit extract a common function in `zerver/lib/stream.py` called `get_stream_by_narrow_operand_access_unchecked` be called by both this code path and the `by_stream` code path, which would let us clean up the linter situation too.

Reply:
Did both of these things. For the `get_stream_from_narrow` I named it `get_stream_from_narrow_access_unchecked` since that what it does.